transformer_engine/common/comm_gemm_overlap/userbuffers/userbuffers-host.cpp

```diff
@@ -280,7 +280,7 @@ int create_communicator_grouped2(communicator **comm, int myrank, int numranks,
       int fd;
       volatile uint32_t abortFlag = 0;
       IpcSocketHandle ipcSock = {0};
-      uint64_t opId = 0xdeadcafeb000 + (*comm)->my_node + (*comm)->ar2_firstgpu;
+      uint64_t opId = 0xdeadcafe0000 + (*comm)->my_node;
```

Comment:
I'm confused how this is related to https://github.com/NVIDIA/TransformerEngine/pull/1470. From what I can tell, we use random hashes to coordinate socket connections in `create_communicator_grouped2` and `register_user_buffer_collective`. These connections are opened, but seemingly never closed? This PR changes `create_communicator_grouped2` and `register_user_buffer_collective` so that they use the same hash (so they can reuse the socket connection?). It doesn't seem that this logic has been touched in a while, so is this change orthogonal to the actual bugfix?

Reply:
The hash cleanup changes are orthogonal. I can remove them if you want.

1 - With #1470, `(*comm)->ar2_firstgpu` is always zero, so I removed it from the hash.
2- Imodify the base of the hash `0xdeadcafe0000` to start with zeros so that `(*comm)->my_node` is identifiable in the socket name.

socket is closed each time after memory handle exchange (see `ipcSocketClose(&ipcSock)`) so there is no conflict by a call to `create_communicator_grouped2` followed by repeated calls of `register_user_buffer_collective`.

Reply:
Makes sense to me, just wanted to understand. They seem like reasonable changes.